File: example/kotlinlib/android/2-jetpack-compose-hello-world/app/src/main/kotlin/com/helloworld/app/MainActivity.kt
package com.helloworld.app

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            Greeting("Hello, World!")
        }
    }
}

@Composable
fun Greeting(name: String) {
    Text(text = name)
}

File: example/androidlib/scala/1-hello-scala/app/src/main/scala/com/helloworld/app/MainActivity.scala
package com.helloworld.app

import android.app.Activity
import android.os.Bundle
import android.widget.TextView
import com.google.gson.Gson

case class Greeting(message: String, version: Int)

object MainActivity {
  private val gson = Gson()

  def parseMessage(json: String): String = {
    val g = gson.fromJson(json, classOf[Greeting])
    g.message
  }
}

class MainActivity extends Activity {
  override def onCreate(savedInstanceState: Bundle): Unit = {
    super.onCreate(savedInstanceState)

    // Parse message using Gson
    val json = getString(R.string.json_greeting)
    val message = MainActivity.parseMessage(json)

    // Create a new TextView
    val textView = TextView(this)
    textView.setText(message)

    setContentView(textView)
  }
}
